flexmeasures/data/schemas/times.py

```diff
@@ -15,6 +15,39 @@ class DurationValidationError(FMValidationError):
     status = "INVALID_PERIOD"  # USEF error status
 
 
+class NewDurationField(MarshmallowClickMixin, fields.Str):
+    """Field that deserializes to a ISO8601 timedelta.
+    and serializes back to an ISO8601 string."""
+
+    def _deserialize(self, value, attr, obj, **kwargs) -> str:
+        """
+        Use the isodate library to validate an ISO8601 string.
+        This method throws a ValidationError if the string is not ISO norm
+        or if the timedelta is able to be represented in multiples of
+        minutes.
+        """
+        try:
+            value_isodate = isodate.parse_duration(value)
+        except ISO8601Error as iso_err:
+            raise DurationValidationError(
+                f"Cannot parse {value} as ISO8601 duration: {iso_err}"
+            )
+
+        if value_isodate.seconds % 60 != 0 or value_isodate.microseconds != 0:
+            raise DurationValidationError(
+                "FlexMeasures only support multiples of 1 minute."
+            )
+
+        return value
```

Comment:
We are returning the original value here (and not `value_isodate`).
Should we change this to value_isodate? Why is no test failing if we should?

If all we do here is to validate (and not returning something else than we got, we could make this a validation function.

Reply:
This was part of the NewDurationField that has been removed.